test/current-item.test.js

```diff
@@ -0,0 +1,187 @@
+import QUnit from 'qunit';
```

Comment:
This should be a relatively robust set of tests for the behavior of the `currentItem` function. This was the first thing I did. The workflow I took for this PR was:

1. Revert the `currentItem` changes
1. Write these tests
1. Restore the `currentItem` changes (source code, not tests)
1. Fix what broke